Lib/test/test_capi/test_file.py

```diff
@@ -220,6 +229,23 @@ def __repr__(self):
             "<str>",
         )
 
+    def test_file_write_custom_obj_raises(self):
+        class ReprRaises:
+            def __repr__(self):
+                raise ValueError("repr raised")
+
+        with self.assertRaisesRegex(ValueError, "repr raised"):
+            self.write_and_return(ReprRaises())
+        with self.assertRaisesRegex(ValueError, "repr raised"):
+            self.write_and_return(ReprRaises(), flags=_testcapi.Py_PRINT_RAW)
+
+        class StrRaises:
+            def __str__(self):
+                raise ValueError("str raised")
+
+        with self.assertRaisesRegex(ValueError, "str raised"):
+            self.write_and_return(StrRaises(), flags=_testcapi.Py_PRINT_RAW)
```

Comment:
It is not clear what is the difference between these tests if it raises in any case. You should either define `__str__` and `__repr__` that do not raise in corresponding classes and test both classes with and without Py_PRINT_RAW, or just make both `__str__` and `__repr__` in the same class raising different exceptions and test that writing with and without Py_PRINT_RAW gives different errors. The former option will duplicate other tests, so I suggest the later way.

Oh, and you do not need to use write_and_return here.